shell/components/form/ServicePorts.vue

```diff
@@ -27,6 +28,12 @@ export default {
     autoAddIfEmpty: {
       type:    Boolean,
       default: true,
+    },
+    rules: {
+      default:   () => [],
+      type:      Array,
+      // we only want functions in the rules array
+      validator: rules => rules.every(rule => ['function'].includes(typeof rule))
```

Comment:
I dont think is reliable and we should definitely do not duplicate code.

Reply:
I don't see the failure point but I agree it's cleaner in TS than with this function. Unfortunately, converting a component of this size to TS just to get cleaner prop validation is probably beyond the scope of this PR at this time.